File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/roadrunnerext/geometry/Circle.java
package org.firstinspires.ftc.teamcode.roadrunnerext.geometry;

import com.acmerobotics.roadrunner.geometry.Vector2d;

public class Circle {
    Vector2d center;
    double radius;

    public Circle(Vector2d center, double radius) {
        this.center = center;
        this.radius = radius;
    }

    public Vector2d getCenter() {
        return center;
    }

    public double getRadius() {
        return radius;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/roadrunnerext/geometry/Circle.kt
package org.firstinspires.ftc.teamcode.roadrunnerext.geometry

import com.acmerobotics.roadrunner.geometry.Vector2d

class Circle(var center: Vector2d, var radius: Double) {
    fun expandedRadius(radiusOffset: Double) = Circle(center, radius + radiusOffset)
    fun getTangentPointToPoint(point: Vector2d) {
        // y = mx + b
        // (x - center.x)^2 + (y - center.y)^2 = r^2
        // (x - center.x)^2 + ((mx + b)- center.y)^2 = r^2
        // x^2 + center.x^2 - 2 * x * center.x
    }
}